### PR TITLE
chore(php): manage php8 compatibility

### PR DESCRIPTION
**Pull request by kduret**
## Description

manage php8 compatibility

**Fixes** MON-7343

## Type of change

- [x] Patch fixing an issue (non-breaking change)
- [ ] New functionality (non-breaking change)
- [ ] Breaking change (patch or feature) that might cause side effects breaking part of the Software
- [ ] Updating documentation (missing information, typo...)

## Target serie

- [ ] 20.04.x
- [ ] 20.10.x
- [ ] 21.04.x
- [x] 21.10.x (master)

### CODE REVIEW

File: Jenkinsfile [hunk 1]
```diff
@@ -314,7 +314,7 @@ try {
 
   stage('Package') {
     def parallelSteps = [:]
-    def osBuilds = isStableBuild() ? ['centos7', 'centos8'] : ['centos7']
+    def osBuilds = isStableBuild() ? ['centos7', 'centos8'] : ['centos7', 'centos8']
```
Comment (sc979) [inline Jenkinsfile:317]:
```suggestion
    def osBuilds = ['centos7', 'centos8']
```

File: www/include/Administration/parameters/general/form.ihtml [hunk 2]
```diff
@@ -226,7 +226,11 @@
     </tr>
     <tr class="list_one"><td class="FormRowField"><img class="helpTooltip" name="tip_send_statistics">{$form.send_statistics.label}</td><td class="FormRowValue">{$form.send_statistics.html}</td></tr>
 
-    <tr class="list_lvl_1"><td class="ListColLvl1_name" colspan="2">{$form.required._note}</td></tr>
+    <tr class="list_lvl_1"><td class="ListColLvl1_name" colspan="2">
+                {if isset($form.required)}
+                    {$form.required._note}
+                {/if}
+            </td></tr>
```
Comment (sc979) [inline www/include/Administration/parameters/general/form.ihtml:233]:
```suggestion
        {if isset($form.required)}
            {$form.required._note}
        {/if}
        </td></tr>
```

File: www/include/Administration/parameters/ldap/form.ihtml [hunk 3]
```diff
@@ -68,7 +68,11 @@
 	<tr class="list_one" id="group_filter"><td class="FormRowField"><img class="helpTooltip" name="group_filter" /><span>{$form.group_filter.label}</span></td><td class="FormRowValue">{$form.group_filter.html}</td></tr>
 	<tr class="list_two" id="group_name"><td class="FormRowField"><img class="helpTooltip" name="group_name" /><span>{$form.group_name.label}</span></td><td class="FormRowValue">{$form.group_name.html}</td></tr>
 	<tr class="list_one" id="group_member"><td class="FormRowField"><img class="helpTooltip" name="group_member" />{$form.group_member.label}</td><td class="FormRowValue">{$form.group_member.html}</td></tr>
-	<tr class="list_lvl_1"><td class="ListColLvl1_name" colspan="2">{$form.required._note}</td></tr>
+	<tr class="list_lvl_1"><td class="ListColLvl1_name" colspan="2">
+                {if isset($form.required)}
+                    {$form.required._note}
+                {/if}
+            </td></tr>
```
Comment (sc979) [inline www/include/Administration/parameters/ldap/form.ihtml:75]:
```suggestion
	    {if isset($form.required)}
	        {$form.required._note}
	    {/if}
	    </td></tr>
```

File: www/include/configuration/configCentreonBroker/formCentreonBroker.ihtml [hunk 4]
```diff
@@ -106,9 +112,16 @@
 	{/foreach}
 	<div id="validForm">
 	{if $o == "a" || $o == "c"}
-		<p class="oreonbutton">{$form.submitC.html}{$form.submitA.html}&nbsp;&nbsp;&nbsp;{$form.reset.html}</p>
+		<p class="oreonbutton">
```
Comment (sc979) [inline www/include/configuration/configCentreonBroker/formCentreonBroker.ihtml:115]:
```suggestion
	    <p class="oreonbutton">
```

File: www/include/configuration/configKnowledge/templates/pagination.ihtml [hunk 5]
```diff
@@ -1,15 +1,17 @@
...
-	{if $pageArr[$key].num != $num}
-	&nbsp;<a href="{$pageArr[$key].url_page}{if $host_name}&host_name={$host_name}{/if}" class="otherPageNumber">{$pageArr[$key].label_page}</a>
-	{else}
-	&nbsp;<b class="currentPageNumber">{$pageArr[$key].label_page}</b>
+	{if isset($firstPage)}&nbsp;<a href="{$firstPage}{if isset($host_name)}&host_name={$host_name}{/if}"><img src="./img/icons/first_rewind.png" title='{$first}'></a>{/if}
+	{if isset($pagePrev)}&nbsp;<a href="{$pagePrev}{if isset($host_name)}&host_name={$host_name}{/if}"><img src="./img/icons/rewind.png" title='{$previous}'></a>{/if}
+	{if isset($pageArr)}
+		{foreach key=key item=item from=$pageArr }
+		{if $pageArr[$key].num != $num}
+		&nbsp;<a href="{$pageArr[$key].url_page}{if isset($host_name)}&host_name={$host_name}{/if}" class="otherPageNumber">{$pageArr[$key].label_page}</a>
```
Comment (sc979) [inline www/include/configuration/configKnowledge/templates/pagination.ihtml:7]:
```suggestion
		    &nbsp;<a href="{$pageArr[$key].url_page}{if isset($host_name)}&host_name={$host_name}{/if}" class="otherPageNumber">{$pageArr[$key].label_page}</a>
```

File: www/include/configuration/configObject/contact/formContactLight.ihtml [hunk 6]
```diff
@@ -83,14 +96,26 @@
...
+                    {$form.required._note}
+                {/if}
+            </td></tr>
         {/if}
     </table>
     <div id="validForm">
     {if $o == "a" || $o == "c" || $o == "mc"}
-        <p class="oreonbutton">{$form.submitC.html}{$form.submitMC.html}{$form.submitA.html}&nbsp;&nbsp;&nbsp;{$form.reset.html}</p>
+        <p class="oreonbutton">
+{if isset($form.submitC)}
```
Comment (sc979) [inline www/include/configuration/configObject/contact/formContactLight.ihtml:109]: wrong indentation

File: www/include/configuration/configObject/contact_template_model/formContactTemplateModel.ihtml [hunk 7]
```diff
@@ -94,15 +107,27 @@
...
+                {/if}
+            </td></tr>
         {/if}
     </table>
     </div>
     <div id="validForm">
     {if $o == "a" || $o == "c" || $o == "mc"}
-        <p class="oreonbutton">{$form.submitC.html}{$form.submitMC.html}{$form.submitA.html}&nbsp;&nbsp;&nbsp;{$form.reset.html}</p>
+        <p class="oreonbutton">
+{if isset($form.submitC)}
```
Comment (sc979) [inline www/include/configuration/configObject/contact_template_model/formContactTemplateModel.ihtml:121]: wrong indentation

File: www/include/configuration/configObject/host/listHost.php [hunk 8]
```diff
@@ -414,7 +414,7 @@
     array(
         "addL" => "main.php?p=" . $p . "&o=a",
         "addT" => _("Add"),
-        "delConfirm" => _("Do you confirm the deletion ?")
+        "delConfirm" => _("Do you confirm the deletion ?"),
```
Comment (sc979) [inline www/include/configuration/configObject/host/listHost.php:417]: why adding the comma ?

File: www/include/configuration/configObject/hostgroup_dependency/DB-Func.php [hunk 9]
```diff
@@ -199,8 +199,8 @@ function insertHostGroupDependency($ret = array()): int
     $statement->bindValue(':depName', $resourceValues['dep_name'], \PDO::PARAM_STR);
     $statement->bindValue(':depDescription', $resourceValues['dep_description'], \PDO::PARAM_STR);
     $statement->bindValue(':inheritsParent', $resourceValues['inherits_parent'], \PDO::PARAM_STR);
-    $statement->bindValue(':executionFailure', $resourceValues['execution_failure_criteria'], \PDO::PARAM_STR);
-    $statement->bindValue(':notificationFailure', $resourceValues['notification_failure_criteria'], \PDO::PARAM_STR);
+    $statement->bindValue(':executionFailure', $resourceValues['execution_failure_criteria'] ?? null, \PDO::PARAM_STR);
+    $statement->bindValue(':notificationFailure', $resourceValues['notification_failure_criteria'] ?? null, \PDO::PARAM_STR);
```
Comment (sc979) [inline www/include/configuration/configObject/hostgroup_dependency/DB-Func.php:203]:
```suggestion
    $statement->bindValue(
        ':executionFailure',
        $resourceValues['execution_failure_criteria'] ?? null,
        \PDO::PARAM_STR
    );
    $statement->bindValue(
        ':notificationFailure',
        $resourceValues['notification_failure_criteria'] ?? null,
        \PDO::PARAM_STR
    );
```

File: www/include/configuration/configObject/service/formService.ihtml [hunk 10]
```diff
@@ -382,18 +404,29 @@
 </div>
 <div id="validForm">
 {if $o == "a" || $o == "c" || $o == "mc"}
-    <p class="oreonbutton">{$form.submitC.html}{$form.submitMC.html}{$form.submitA.html}&nbsp;&nbsp;&nbsp;{$form.reset.html}</p>
+    <p class="oreonbutton">
+{if isset($form.submitC)}
```
Comment (sc979) [inline www/include/configuration/configObject/service/formService.ihtml:408]: wrong indentation

File: www/include/configuration/configObject/service_categories/formServiceCategories.ihtml [hunk 11]
```diff
@@ -49,9 +57,17 @@
 	</div>
 	<div id="validForm">
 	{if $o == "a" || $o == "c" || $o == "mc"}
-		<p class="oreonbutton">{$form.submitC.html}{$form.submitMC.html}{$form.submitA.html}&nbsp;&nbsp;&nbsp;{$form.reset.html}</p>
+		<p class="oreonbutton">
+{if isset($form.submitC)}
```
Comment (sc979) [inline www/include/configuration/configObject/service_categories/formServiceCategories.ihtml:61]: wrong indentation

File: www/include/configuration/configObject/traps/formTraps.ihtml [hunk 12]
```diff
@@ -9,9 +8,15 @@
         </ul>
         <div id="validFormTop">
 	     {if $o == "a" || $o == "c"}
-		    <p class="oreonbutton">{$form.submitC.html}{$form.submitA.html}&nbsp;&nbsp;&nbsp;{$form.reset.html}</p>
+		    <p class="oreonbutton">
+      {if isset($form.submitC)}
```
Comment (sc979) [inline www/include/configuration/configObject/traps/formTraps.ihtml:12]: wrong indentation

File: www/include/configuration/configResources/formResources.ihtml [hunk 13]
```diff
@@ -2,9 +2,15 @@
 <form {$form.attributes}>
 	<div id="validFormTop">
 	{if $o == "a" || $o == "c"}
-		<p class="oreonbutton">{$form.submitC.html}{$form.submitA.html}&nbsp;&nbsp;&nbsp;{$form.reset.html}</p>
+		<p class="oreonbutton">
+{if isset($form.submitC)}
```
Comment (sc979) [inline www/include/configuration/configResources/formResources.ihtml:6]: wrong indentation

File: www/include/configuration/configResources/formResources.ihtml [hunk 14]
```diff
@@ -44,15 +50,25 @@
...
+                {/if}
+            </td></tr>
 		{/if}
 	</table>
 	</div>
 	<div id="validForm">
 	{if $o == "a" || $o == "c"}
-		<p class="oreonbutton">{$form.submitC.html}{$form.submitA.html}&nbsp;&nbsp;&nbsp;{$form.reset.html}</p>
+		<p class="oreonbutton">
+{if isset($form.submitC)}
```
Comment (sc979) [inline www/include/configuration/configResources/formResources.ihtml:64]: wrong indentation

File: www/include/eventLogs/template/viewLog.ihtml [hunk 15]
```diff
@@ -66,8 +66,20 @@
         </tr>
         <tr>
             <td class="log-select">{$form.output.html}</td>
-            <td>{$form.StartDate.html}{$form.alternativeDateStartDate.html}{$form.StartTime.html}</td>
-            <td>{$form.EndDate.html}{$form.alternativeDateEndDate.html}{$form.EndTime.html}</td>
+            <td>
+                {$form.StartDate.html}
+                {if isset($form.alternativeDateStartDate)}
```
Comment (sc979) [inline www/include/eventLogs/template/viewLog.ihtml:71]: this one is mandatory to be able to localize the date.
Shouldn't you set it if not set ?

File: www/include/eventLogs/template/viewLogEngine.ihtml [hunk 16]
```diff
@@ -25,12 +25,16 @@
                 <h4>{t}From{/t}</h4>
                 {$form.StartDate.html}
                 {$form.StartTime.html}
-                {$form.alternativeDateStartDate.html}
+                {if isset($form.alternativeDateStartDate)}
```
Comment (sc979) [inline www/include/eventLogs/template/viewLogEngine.ihtml:28]: this one is mandatory to be able to localize the date.
Shouldn't you set it if not set ?

File: www/include/monitoring/downtime/template/AddDowntime.ihtml [hunk 17]
```diff
@@ -124,12 +117,20 @@
...
+            {/if}
+            &nbsp;{$form.start_time.html}
             </td>
         </tr>
         <tr class="list_one">
             <td class="FormRowField">{$form.end.label}</td>
-            <td class="FormRowValue">{$form.end.html}&nbsp;{$form.alternativeDateEnd.html}&nbsp;{$form.end_time.html}
+            <td class="FormRowValue">{$form.end.html}&nbsp;
+            {if isset($form.alternativeDateEnd)}
+                {$form.alternativeDateEnd.html}
```
Comment (sc979) [inline www/include/monitoring/downtime/template/AddDowntime.ihtml:131]: this one is mandatory to be able to localize the date.
Shouldn't you set it if not set ?

File: www/include/reporting/dashboard/ajaxReporting_js.php [hunk 18]
```diff
@@ -41,6 +41,9 @@
     $arg = "id=" . $id;
 }
 
+$startDate = 0;
+$endDate = 0;
```
Comment (sc979) [inline www/include/reporting/dashboard/ajaxReporting_js.php:45]: aren't they already set from the script calling this one ?

File: www/include/reporting/dashboard/template/viewHostLog.ihtml [hunk 19]
```diff
@@ -24,8 +22,14 @@
 				<input type="radio" value="preset" name="period_choice" {if $period_choice == "preset" } checked="checked" {/if}/>
 					{$periodTitle} : {$formPeriod.period.html}{$periodORlabel}
 				<input type="radio" name="period_choice" value="custom" {if $period_choice == "custom" } checked="checked" {/if} />
-					{$formPeriod.StartDate.label} {$formPeriod.StartDate.html}&nbsp;{$formPeriod.alternativeDateStartDate.html}
-					{$formPeriod.EndDate.label} {$formPeriod.EndDate.html} {$formPeriod.alternativeDateEndDate.html}
+					{$formPeriod.StartDate.label} {$formPeriod.StartDate.html}&nbsp;
+					{if isset($formPeriod.alternativeDateStartDate)}
+						{$formPeriod.alternativeDateStartDate.html}
```
Comment (sc979) [inline www/include/reporting/dashboard/template/viewHostLog.ihtml:27]: this one is mandatory to be able to localize the date.
Shouldn't you set it if not set ?

File: Jenkinsfile (hunk 1 above)
Comment (kduret) [reply Jenkinsfile:317]: we'll revert this juste before merge, but we let it for testing purpose

File: unattended.sh [hunk 20]
```diff
@@ -317,12 +320,15 @@ function set_required_prerequisite() {
 			;;
 		esac
 		RELEASE_RPM_URL="http://yum.centreon.com/standard/$CENTREON_MAJOR_VERSION/el7/stable/noarch/RPMS/centreon-release-$CENTREON_RELEASE_VERSION.el7.centos.noarch.rpm"
+        REMI_RELEASE_RPM_URL="https://rpms.remirepo.net/enterprise/remi-release-8.rpm"
```
[the remark this answers is not in the document]
Comment (kduret) [reply unattended.sh:323]: We should change other lines, not this one, cause indent are tabs

[the remark this answers is not in the document]
Comment (kduret) [reply unattended.sh:323]: i'll do it

File: www/class/centreon-clapi/centreonAPI.class.php [hunk 21]
```diff
@@ -346,7 +346,9 @@ function ($n) {
          * Manage version
          */
         $this->optGen = $this->getOptGen();
-        $version = $this->optGen["version"];
+        if (is_array($this->optGen)) {
+            $version = $this->optGen["version"];
```
[the remark this answers is not in the document]
Comment (kduret) [reply www/class/centreon-clapi/centreonAPI.class.php:350]: in fact, we should delete it, cause `$version` seems not used xD

File: unattended.sh [hunk 22]
```diff
@@ -317,12 +320,15 @@ function set_required_prerequisite() {
...
 		log "INFO" "Install Centreon from ${RELEASE_RPM_URL}"
-		PHP_BIN="/opt/rh/rh-php73/root/bin/php"
-		PHP_ETC="/etc/opt/rh/rh-php73/php.d/"
```
[the remark this answers is not in the document]
Comment (tuntoja) [reply unattended.sh:322]: the 71 and 72 are still used.
lines 321 / 322 are for centos7 only
however, if there is a syntax/option specific for php in centos8, it should be handled indeed
not sure what L675 and 685 are though ? (one is empty, other is a comment)

File: www/class/centreon-clapi/centreonAPI.class.php (hunk 21 above)
[the remark this answers is not in the document]
Comment (kduret) [reply www/class/centreon-clapi/centreonAPI.class.php:350]: it is used as global :'(

File: unattended.sh (hunk 22 above)
[the remark this answers is not in the document]
Comment (kduret) [reply unattended.sh:322]: I moved these variables a the top, cause they have the same value on rhel7 & rhel8

File: www/include/Administration/myAccount/formMyAccount.ihtml [hunk 23]
```diff
@@ -104,15 +113,26 @@
             <tr class="list_one"><td class="FormRowField">{$form.monitoring_sound_svc_notification_2.label}</td><td class="FormRowValue">{$form.monitoring_sound_svc_notification_2.html}</td></tr>
             <tr class="list_two"><td class="FormRowField">{$form.monitoring_sound_svc_notification_3.label}</td><td class="FormRowValue">{$form.monitoring_sound_svc_notification_3.html}</td></tr>
     		{if $o == "c"}
-    		<tr class="list_lvl_1"><td class="ListColLvl1_name" colspan="2">{$form.required._note}</td></tr>
+    		<tr class="list_lvl_1"><td class="ListColLvl1_name" colspan="2">
+                {if isset($form.required)}
+                    {$form.required._note}
+                {/if}
+            </td></tr>
```
[the remark this answers is not in the document]
Comment (kduret) [reply www/include/Administration/myAccount/formMyAccount.ihtml:120]: this is the whole file which use tab and space indents...
So I can't really fix it except my updating the entire file

File: www/include/configuration/configObject/host/listHost.php (hunk 8 above)
Comment (kduret) [reply www/include/configuration/configObject/host/listHost.php:417]: I'll remove it

File: www/include/reporting/dashboard/ajaxReporting_js.php (hunk 18 above)
Comment (kduret) [reply www/include/reporting/dashboard/ajaxReporting_js.php:45]: not every time, but I moved those cause it was overrinding post values

File: www/include/configuration/configKnowledge/pagination.php [hunk 24]
```diff
@@ -269,7 +270,9 @@ function setL(_this) {
 $tpl->assign("begin", $num);
 $tpl->assign("end", $limit);
 $tpl->assign("pagin_page", _("Page"));
-$tpl->assign("order", $_GET["order"]);
+if (isset($_GET["order"])) {
```
[the remark this answers is not in the document]
Comment (sc979) [reply www/include/configuration/configKnowledge/pagination.php:273]: this one should also be sanitized using a whitelist for example (if !== 'DESC', then = 'ASC')

File: www/class/centreonCfgWriter.class.php [hunk 25]
```diff
@@ -39,7 +39,6 @@
 class CentreonCfgWriter
 {
     private $buffer;
-    private $fd;
```
Comment (a-launois) [inline www/class/centreonCfgWriter.class.php:42]: shouldn't L59 be commented too ?

File: www/include/monitoring/downtime/template/AddDowntime.ihtml (hunk 17 above)
Comment (kduret) [reply www/include/monitoring/downtime/template/AddDowntime.ihtml]: I removed totally this check, cause alternative dates are already set by `$form.hidden`

File: www/class/centreonCfgWriter.class.php (hunk 25 above)
Comment (kduret) [reply www/class/centreonCfgWriter.class.php:42]: Indeed :)
I've removed it

File: www/include/configuration/configNagios/formNagios.ihtml [hunk 26]
```diff
@@ -12,9 +12,16 @@
...
         {if $o == "a" || $o == "c"}
-            <p class="oreonbutton">{$form.submitC.html}{$form.submitA.html}&nbsp;&nbsp;&nbsp;{$form.reset.html}</p>
+            <p class="oreonbutton">
+				{if isset($form.submitC)}
+					{$form.submitC.html}
+				{else}
+					{$form.submitA.html}
+				{/if}
+				&nbsp;&nbsp;&nbsp;{$form.reset.html}
+			</p>
```
Comment (a-launois) [inline www/include/configuration/configNagios/formNagios.ihtml:22]:
```suggestion
            <p class="oreonbutton">
               {if isset($form.submitC)}
                   {$form.submitC.html}
               {else}
                   {$form.submitA.html}
               {/if}
                   &nbsp;&nbsp;&nbsp;{$form.reset.html}
           </p>
```

File: www/include/Administration/parameters/knowledgeBase/formKnowledgeBase.php [hunk 27]
```diff
@@ -44,7 +44,9 @@
 while ($opt = $DBRESULT->fetchRow()) {
     $gopt[$opt["key"]] = myDecode($opt["value"]);
 }
-$gopt[$opt["kb_wiki_password"]] = CentreonAuth::PWS_OCCULTATION;
+if (isset($gopt["kb_wiki_password"])) {
```
Comment (a-launois) [inline www/include/Administration/parameters/knowledgeBase/formKnowledgeBase.php:47]:
```suggestion
if (isset($opt["kb_wiki_password"])) {
```

File: www/include/configuration/configNagios/formNagios.ihtml [hunk 28]
```diff
@@ -74,7 +81,11 @@
             </tr>
 			<tr class="list_one"><td class="FormRowField">&nbsp;{$form.nagios_comment.label}</td><td class="FormRowValue">{$form.nagios_comment.html}</td></tr>
 			{if $o == "a" || $o == "c"}
-				<tr class="list_lvl_1"><td class="ListColLvl1_name" colspan="2">{$form.required._note}</td></tr>
+				<tr class="list_lvl_1"><td class="ListColLvl1_name" colspan="2">
+                {if isset($form.required)}
+                    {$form.required._note}
+                {/if}
+            </td></tr>
```
Comment (a-launois) [inline www/include/configuration/configNagios/formNagios.ihtml:88]:
```suggestion
				<tr class="list_lvl_1"><td class="ListColLvl1_name" colspan="2">
				{if isset($form.required)}
 					{$form.required._note}
				{/if}
				</td></tr>
```

File: www/include/configuration/configNagios/formNagios.ihtml [hunk 29]
```diff
@@ -163,7 +174,11 @@
 			<tr class="list_two"><td class="FormRowField"><img class="helpTooltip" name="passive_host_checks_are_soft">&nbsp;{$form.passive_host_checks_are_soft.label}</td><td class="FormRowValue">{$form.passive_host_checks_are_soft.html}</td></tr>
 
 			{if $o == "a" || $o == "c"}
-				<tr class="list_lvl_1"><td class="ListColLvl1_name" colspan="2">{$form.required._note}</td></tr>
+				<tr class="list_lvl_1"><td class="ListColLvl1_name" colspan="2">
+                {if isset($form.required)}
+                    {$form.required._note}
+                {/if}
+            </td></tr>
```
Comment (a-launois) [inline www/include/configuration/configNagios/formNagios.ihtml:181]:
```suggestion
				<tr class="list_lvl_1"><td class="ListColLvl1_name" colspan="2">
				{if isset($form.required)}
					{$form.required._note}
				{/if}
 				</td></tr>
```

File: www/include/configuration/configNagios/formNagios.ihtml [hunk 30]
```diff
@@ -229,7 +244,11 @@
                         <tr class="list_two"><td class="FormRowField"><img class="helpTooltip" name="retained_service_attribute_mask">&nbsp;{$form.retained_service_attribute_mask.label}</td><td class="FormRowValue">{$form.retained_service_attribute_mask.html}</td></tr>
 
 			{if $o == "a" || $o == "c"}
-				<tr class="list_lvl_1"><td class="ListColLvl1_name" colspan="2">{$form.required._note}</td></tr>
+				<tr class="list_lvl_1"><td class="ListColLvl1_name" colspan="2">
+                {if isset($form.required)}
+                    {$form.required._note}
+                {/if}
+            </td></tr>
```
Comment (a-launois) [inline www/include/configuration/configNagios/formNagios.ihtml:251]:
```suggestion
				<tr class="list_lvl_1"><td class="ListColLvl1_name" colspan="2">
 				{if isset($form.required)}
					{$form.required._note}
				{/if}
				</td></tr>
```

File: www/include/configuration/configNagios/formNagios.ihtml [hunk 31]
```diff
@@ -295,7 +314,11 @@
 			<tr class="list_two"><td class="FormRowField"><img class="helpTooltip" name="host_perfdata_file_processing_command">&nbsp;{$form.host_perfdata_file_processing_command.label}</td><td class="FormRowValue">{$form.host_perfdata_file_processing_command.html}</td></tr>
 			<tr class="list_one"><td class="FormRowField"><img class="helpTooltip" name="service_perfdata_file_processing_command">&nbsp;{$form.service_perfdata_file_processing_command.label}</td><td class="FormRowValue">{$form.service_perfdata_file_processing_command.html}</td></tr>
 			{if $o == "a" || $o == "c"}
-				<tr class="list_lvl_1"><td class="ListColLvl1_name" colspan="2">{$form.required._note}</td></tr>
+				<tr class="list_lvl_1"><td class="ListColLvl1_name" colspan="2">
+                {if isset($form.required)}
+                    {$form.required._note}
+                {/if}
+            </td></tr>
```
Comment (a-launois) [inline www/include/configuration/configNagios/formNagios.ihtml:321]:
```suggestion
				<tr class="list_lvl_1"><td class="ListColLvl1_name" colspan="2">
				{if isset($form.required)}
					{$form.required._note}
				{/if}
				</td></tr>
```

File: www/include/configuration/configNagios/formNagios.ihtml [hunk 32]
```diff
@@ -359,7 +382,11 @@
 			<tr class="list_two"><td class="FormRowField"><img class="helpTooltip" name="enable_environment_macros">&nbsp;{$form.enable_environment_macros.label}</td><td class="FormRowValue">{$form.enable_environment_macros.html}</td></tr>
             <tr class="list_one"><td class="FormRowField"><img class="helpTooltip" name="use_setpgid">&nbsp;{$form.use_setpgid.label}</td><td class="FormRowValue">{$form.use_setpgid.html}</td></tr>
 			{if $o == "a" || $o == "c"}
-				<tr class="list_lvl_1"><td class="ListColLvl1_name" colspan="2">{$form.required._note}</td></tr>
+				<tr class="list_lvl_1"><td class="ListColLvl1_name" colspan="2">
+                {if isset($form.required)}
+                    {$form.required._note}
+                {/if}
+            </td></tr>
```
Comment (a-launois) [inline www/include/configuration/configNagios/formNagios.ihtml:389]:
```suggestion
				<tr class="list_lvl_1"><td class="ListColLvl1_name" colspan="2">
				{if isset($form.required)}
					{$form.required._note}
				{/if}
				</td></tr>
```

File: www/include/configuration/configNagios/formNagios.ihtml [hunk 33]
```diff
@@ -384,7 +411,11 @@
 			<tr class="list_one"><td class="FormRowField"><img class="helpTooltip" name="admin_email">&nbsp;{$form.admin_email.label}</td><td class="FormRowValue">{$form.admin_email.html}</td></tr>
 			<tr class="list_two"><td class="FormRowField"><img class="helpTooltip" name="admin_pager">&nbsp;{$form.admin_pager.label}</td><td class="FormRowValue">{$form.admin_pager.html}</td></tr>
 			{if $o == "a" || $o == "c"}
-				<tr class="list_lvl_1"><td class="ListColLvl1_name" colspan="2">{$form.required._note}</td></tr>
+				<tr class="list_lvl_1"><td class="ListColLvl1_name" colspan="2">
+                {if isset($form.required)}
+                    {$form.required._note}
+                {/if}
+            </td></tr>
```
Comment (a-launois) [inline www/include/configuration/configNagios/formNagios.ihtml:418]:
```suggestion
				<tr class="list_lvl_1"><td class="ListColLvl1_name" colspan="2">
				{if isset($form.required)}
					{$form.required._note}
				{/if}
				</td></tr>
```

File: www/include/configuration/configNagios/formNagios.ihtml [hunk 34]
```diff
@@ -406,15 +437,26 @@
 			<tr class="list_two"><td class="FormRowField"><img class="helpTooltip" name="max_debug_file_size">&nbsp;{$form.max_debug_file_size.label}</td><td class="FormRowValue">{$form.max_debug_file_size.html}&nbsp;{$Bytes}</td></tr>
 			<tr class="list_one"><td class="FormRowField"><img class="helpTooltip" name="daemon_dumps_core">&nbsp;{$form.daemon_dumps_core.label}</td><td class="FormRowValue">{$form.daemon_dumps_core.html}&nbsp;</td></tr>
 			{if $o == "a" || $o == "c"}
-				<tr class="list_lvl_1"><td class="ListColLvl1_name" colspan="2">{$form.required._note}</td></tr>
+				<tr class="list_lvl_1"><td class="ListColLvl1_name" colspan="2">
+                {if isset($form.required)}
+                    {$form.required._note}
+                {/if}
+            </td></tr>
```
Comment (a-launois) [inline www/include/configuration/configNagios/formNagios.ihtml:444]:
```suggestion
				<tr class="list_lvl_1"><td class="ListColLvl1_name" colspan="2">
				{if isset($form.required)}
					{$form.required._note}
				{/if}
				</td></tr>
```

**Review by adr-mo — APPROVED** [review]

**Review by jeremyjaouen — APPROVED** [review]

**Review by a-launois — APPROVED** [review]

**Pull request merged**